### PR TITLE
Declare Bio.Crystal obsolete

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #...

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This work might not be necessary #2223 
I think this can go from obsolete to deprecated soon as there is no implementation of the classes.

### CODE REVIEW

**Comment by peterjc** [comment]
This is intended to deprecate ``Bio.Crystal``. Doing this would indirectly solve #2223 (we don't need to dual license ``Bio.Crystal`` if we remove it).

As per https://biopython.org/wiki/Deprecation_policy the email to the mailing list about this potential deprecation is here:

https://mailman.open-bio.org/pipermail/biopython/2019-September/016678.html

File: Bio/Crystal/__init__.py [hunk 1]
```diff
@@ -3,11 +3,37 @@
...
 
-"""Represent the NDB Atlas structure (a minimal subset of PDB format).
+"""Represent the NDB Atlas structure (a minimal subset of PDB format) (OBSOLETE).
 
 Hetero, Crystal and Chain exist to represent the NDB Atlas structure.  Atlas
-is a minimal subset of the PDB format.  Heteo supports a 3 alphameric code.
+is a minimal subset of the PDB format.  Hetero supports a 3 alphanumeric code.
 The NDB web interface is located at http://ndbserver.rutgers.edu
+
+There is no implementation of the classes defined on Bio.Crystal on Biopython,
```
Comment (peterjc) [inline Bio/Crystal/__init__.py]: This sentence does not flow well.

File: Bio/Crystal/__init__.py [hunk 2]
```diff
@@ -61,19 +87,23 @@ def __init__(self, data):
         self.data = data[:].lower()
 
     def __eq__(self, other):
+        """No required."""
```
Comment (peterjc) [inline Bio/Crystal/__init__.py]: What does this mean, and why have you added it to all the methods?

**Comment by peterjc** [comment]
I don't understand what you mean here (from the deprecated file, but also in the docstrings) about "no implementation of the classes": 

> Declare obsolete in release 1.75, there is no implementation of the classes defined on Bio.Crystal.

If you mean nothing else in Biopython depends on ``Bio.Crystal``, that alone is no reason to deprecate something.

> PDB NDB files can be opened with Bio.PDB.

If the NDB files can all be opened with ``Bio.PDB``, that is a good basis to drop ``Bio.Crystal``. Have you confirmed this - or at least a good sample?

Looking at http://ndbserver.rutgers.edu/ndbmodule/download_data/downloadIndex.html they are supporting mmCIF nowadays (good). I would probably focus on testing everything under *NDB only. Files for structures that are present only in NDB and not in PDB.* (a relatively small set).

File: Bio/Crystal/__init__.py (hunk 2 above)
Comment (svalqui) [reply Bio/Crystal/__init__.py]: That relates to #2116 , so we do not have people updating the Docstrings of magic methods when is going to be deprecated.

**Comment by svalqui** [comment]
I meant the module is not used, for instance there is no parser to load the structures. I'll reword it.

**Comment by svalqui** [comment]
Yes, you can use Bio.PDB to open the files
I have tried this file:
http://xray.bmc.uu.se/hicup/001/001_msd.pdb
and this other:
http://ndbserver.rutgers.edu/files/ftp/NDB/coordinates/ndbid_only/na-biol/arb002.pdb1

Both files open and you can see the atoms ::

    from Bio.PDB.PDBParser import PDBParser
    parser = PDBParser(PERMISSIVE=1)
    structure = parser.get_structure("001", "001_msd.pbd")
    for model in structure:
        print('Model ',model)
        for chain in model:
            print('Chain ', chain)
            for residue in chain:
                print('Res ', residue)
                for atom in residue:
                    print('Atom ', atom)

`

File: Bio/Crystal/__init__.py (hunk 2 above)
Comment (peterjc) [reply Bio/Crystal/__init__.py]: Yes, but personally I would say no docstring is better than this unhelpful string applied to all of them. Please either take it out again, or add meaningful docstrings if you prefer.

**Comment by mdehoon** [comment]
+1 on removing Bio.Crystal. This module has no documentation and has not been updated (other than trivial changes) since 2003. Keeping old (and probably unused) modules around makes Biopython look messy, and confusing for beginners.

File: Bio/Crystal/__init__.py [hunk 3]
```diff
@@ -3,11 +3,37 @@
...
-"""Represent the NDB Atlas structure (a minimal subset of PDB format).
+"""Represent the NDB Atlas structure (a minimal subset of PDB format) (OBSOLETE).
 
 Hetero, Crystal and Chain exist to represent the NDB Atlas structure.  Atlas
-is a minimal subset of the PDB format.  Heteo supports a 3 alphameric code.
+is a minimal subset of the PDB format.  Hetero supports a 3 alphanumeric code.
 The NDB web interface is located at http://ndbserver.rutgers.edu
+
+There is no use of the classes defined on Bio.Crystal on Biopython,
+so the module will be removed.
```
Comment (peterjc) [inline Bio/Crystal/__init__.py]: Delete these two lines...

File: Bio/Crystal/__init__.py [hunk 4]
```diff
@@ -3,11 +3,37 @@
...
+        print('Model ',model)
+        for chain in model:
+            print('Chain ', chain)
+            for residue in chain:
+                print('Res ', residue)
+                for atom in residue:
+                    print('Atom ', atom)
+
+Since there is no use of the classes and their function are already
+covered by Bio.PDB, Bio.Crystal will be deprecated.
```
Comment (peterjc) [inline Bio/Crystal/__init__.py]: Replace with: Bio.Crystal is self-contained, with the main functionality covered by Bio.PDB.

File: DEPRECATED.rst [hunk 5]
```diff
@@ -75,6 +75,11 @@ Bio.Index
 ---------
 Deprecated in release 1.75, was not used anywhere in Biopython.
 
+Bio.Crystal
+-----------
+Declare obsolete in release 1.75, there is no implementation of the classes
```
Comment (peterjc) [inline DEPRECATED.rst]: Declare --> Declared

Comment (svalqui) [reply DEPRECATED.rst]: Updated.

File: DEPRECATED.rst [hunk 6]
```diff
@@ -75,6 +75,11 @@ Bio.Index
 ---------
 Deprecated in release 1.75, was not used anywhere in Biopython.
 
+Bio.Crystal
+-----------
+Declared obsolete in release 1.75, there is no implementation of the classes
```
Comment (peterjc) [inline DEPRECATED.rst]: How about just:

Declared obsolete in release 1.75, PDB NDB files can be opened with Bio.PDB.

Comment (svalqui) [reply DEPRECATED.rst]: Done.

**Comment by mdehoon** [comment]
Can we move directly to a BiopythonDeprecationWarning, instead of declaring the module obsolete? Few, if any, users will read the source code and notice that the module was declared obsolete. With a BiopythonDeprecationWarning, at least users (if there are any) will notice.

**Comment by svalqui** [comment]
I'll be offline for a couple of weeks, I'll make any agreed changes then,
Cheers,

**Comment by peterjc** [comment]
That sounds like a plan:

https://mailman.open-bio.org/pipermail/biopython/2019-September/016679.html

Thank you!

**Pull request merged**